File: k-kbk/5주차/3986.py
n = int(input())

words = [input() for _ in range(n)]
result = 0

for word in words:
    stack = []

    for char in word:
        if len(stack) == 0:
            stack.append(char)
            continue

        last = stack[-1]
        if last == char:
            stack.pop()
        else:
            stack.append(char)

    if len(stack) == 0:
        result += 1

print(result)

File: keemsebin/5주차/3986.js
const readline = require("readline");
const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});

let input = [];

rl.on("line", function (line) {
  input.push(line);
}).on("close", function () {
  let N = Number(input[0]);
  let word = input.slice(1).map((item) => item.split(""));
  let result = 0;

  for (let i = 0; i < N; i++) {
    let goodWord = word[i];
    let stack = [];
    for (let j = 0; j < goodWord.length; j++) {
      if (stack.length > 0 && stack[stack.length - 1] === goodWord[j]) {
        stack.pop();
      } else {
        stack.push(goodWord[j]);
      }
    }
    if (stack.length === 0) result++;
  }

  console.log(result);
});
